common/doc.go

```diff
@@ -1,3 +1,28 @@
 // Package common contains wrapper types and utilities common to all of the
 // OOXML document formats.
 package common
+
+import (
+	"archive/zip"
+	"strings"
+
+	"github.com/unidoc/unioffice"
+	"github.com/unidoc/unioffice/zippkg"
+)
+
+//AddImageToZip adds an image (either from bytes or from disk) and adds it to the zip file.
+func AddImageToZip(z *zip.Writer, img ImageRef, imageNum int, dt unioffice.DocType) error {
+	filename := unioffice.AbsoluteImageFilename(dt, imageNum, strings.ToLower(img.Format()))
+	if img.Data() != nil && len(*img.Data()) > 0 {
+		if err := zippkg.AddFileFromBytes(z, filename, *img.Data()); err != nil {
+			return err
+		}
+	} else if img.Path() != "" {
+		if err := zippkg.AddFileFromDisk(z, filename, img.Path()); err != nil {
+			return err
+		}
+	} else {
+		unioffice.Log("unsupported image source: %+v", img)
```

Comment:
Please, make this func return an error instead of logging. Just keep text the same. I think that this func is rather general, it may be used somewhere else and not all the parts of the library may require logging. It's better to log on the client side

Reply:
Good point. Done.